### DISCUSSION TITLE
Powershell - Invoke-Expression: Cannot bind argument to parameter 'Command' because it is an empty string.

### DISCUSSION THREAD

**Discussion by felipecrs** [JanDeDobbeleer/oh-my-posh#458] [Troubleshoot] [answered]
### Prerequisites

- [X] I have read and understand the `CONTRIBUTING` guide
- [X] I looked for duplicate issues before submitting this one

### Description

I installed oh-my-posh with `Install-Module oh-my-posh -AllowPrerelease -Force`, and it fails to run as:

```sh-session
$ Set-PoshPrompt
&: The term 'C:\Users\felipesantos\OneDrive - Instituto Nacional de Telecomunica├º├╡es\Documents\PowerShell\Modules\oh-my-posh\3.97.2\bin\posh-windows-amd64.exe' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
Invoke-Expression: Cannot bind argument to parameter 'Command' because it is an empty string.
```  

It seems to be because the binary path is not right. It was supposed to be:

```
C:\Users\felipesantos\OneDrive - Instituto Nacional de Telecomunicações\Documents\PowerShell\Modules\oh-my-posh\3.97.2\bin\posh-windows-amd64.exe
```

So there must be something wrong while parsing it.

### Environment

- Oh my Posh version: 3.97.2
- Theme: doesn't matter
- Operating System: Windows 10 21H1
- Shell: PowerShell 7.1.2
- Terminal: Windows Terminal Preview 1.6

### Steps to Reproduce

Probably try to install in a system where the Documents folder is backed up with OneDrive for Business, and the company name has special characters.

**Expected behavior:** The commands to work.

**Actual behavior:** No posh commands works.

**Comment by JanDeDobbeleer** [comment, marked as answer]
Tried it on macOS and Windows, works as expected. 

## Windows
<img width="834" alt="image" src="https://user-images.githubusercontent.com/2492783/109198794-ec6c7100-779e-11eb-9bca-e636ddc1cc5a.png">

## macOS
<img width="1718" alt="image" src="https://user-images.githubusercontent.com/2492783/109198871-060db880-779f-11eb-84b4-858a64b14c46.png">

Can you check the encoding in your terminal? As that might have this effect. Ideally PSCore defaults to UTF8 as identified [here](https://github.com/PowerShell/PowerShell/issues/7233). A fix is to add this to your `$PROFILE`: `[console]::InputEncoding = [console]::OutputEncoding = New-Object System.Text.UTF8Encoding`

**Reply by sayagodev** [reply]
I'm new to using the PowerShell and all the technical aspect of terminal on Windows. I had the same problem, thanks @JanDeDobbeleer for resolved.

**Comment by felipecrs** [comment]
![image](https://user-images.githubusercontent.com/29582865/109200233-20379e80-777f-11eb-877c-35564902b591.png)

That works. I'll add it to my profile.

**Comment by felipecrs** [comment]
BTW:

```
IsSingleByte      : True
EncodingName      : OEM United States
WebName           : ibm437
HeaderName        : ibm437
BodyName          : ibm437
Preamble          :
WindowsCodePage   :
IsBrowserDisplay  :
IsBrowserSave     :
IsMailNewsDisplay :
IsMailNewsSave    :
EncoderFallback   : System.Text.InternalEncoderBestFitFallback
DecoderFallback   : System.Text.InternalDecoderBestFitFallback
IsReadOnly        : False
CodePage          : 437
```

**Comment by JanDeDobbeleer** [comment]
The problem is that I could add that to oh-my-posh, but I'm afraid that might introduce other side-effects for people.

**Comment by felipecrs** [comment]
I totally understand. For me, it's ok to keep it in my profile. It doesn't seem the kind of problem to be solved here, but in the Powershell Core itself, most likely.
